src/components/OrganizationAutocomplete/OrganizationAutocomplete.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from 'react';
+import { useDebounce } from 'react-use';
+import { Except } from 'type-fest';
+import { AutocompleteField, AutocompleteFieldProps } from '../form';
+import { useOrganizationAutocompleteQuery } from './OrganizationAutocomplete.generated';
+
+type OrganizationAutocompleteProps<
+  T,
+  Multiple extends boolean | undefined,
+  DisableClearable extends boolean | undefined,
+  FreeSolo extends boolean | undefined
+> = Except<
+  AutocompleteFieldProps<T, Multiple, DisableClearable, FreeSolo>,
+  'options' | 'loading'
+>;
+
+export function OrganizationAutocomplete<
+  T,
+  Multiple extends boolean | undefined,
+  DisableClearable extends boolean | undefined,
+  FreeSolo extends boolean | undefined
+>(
+  props: OrganizationAutocompleteProps<T, Multiple, DisableClearable, FreeSolo>
+) {
+  const [value, setValue] = useState('');
+  const [options, setOptions] = useState<any[]>([]);
+  const [selectedOptions, setSelectedOptions] = useState<any[]>([]);
+  const [searching, setSearching] = useState(false);
+
+  // if we don't want to show any initial options, it'd be nice not to have to call the api before input
+  const { refetch } = useOrganizationAutocompleteQuery({
+    variables: {
+      input: {
+        filter: {
+          name: '',
+        },
+      },
+    },
+  });
+
+  useDebounce(
+    async () => {
+      if (value === '') return;
+      const { data } = await refetch({
+        input: {
+          filter: {
+            name: value,
+          },
+        },
+      });
+      setOptions(
+        data?.organizations?.items.map((item) => item.name.value) || []
+      );
+      setSearching(false);
+    },
+    500,
+    [searching, value]
+  );
+
+  return (
+    <AutocompleteField
+      {...props}
+      hideNoOptionsDiv={!searching && value === ''}
+      options={[...options, ...selectedOptions]}
+      filterSelectedOptions
+      loading={searching}
+      // we do this and set filterSelectedOptions to true ^ to avoid an error in the console about selected options not appearing in the list
+      onOptionSelect={(options: any[]) => {
+        setSelectedOptions(options);
+      }}
+      onInput={(val: any) => {
+        const trimmed = val.trim();
+        // length will be 1 after every add if Autocomplete has multiple
+        // don't want to show stale options from previous input
+        if (trimmed.length <= 1) {
+          setOptions([]);
+        }
+        if (trimmed === '') {
+          setSearching(false);
+        } else {
+          setSearching(true);
+        }
+        setValue(val);
+      }}
```

Comment:
This should also probably be apart of the AutocompleteField so every wrapper doesn't have to implement this.